src/diffusers/loaders.py

```diff
@@ -825,10 +869,18 @@ def load_lora_weights(self, pretrained_model_name_or_path_or_dict: Union[str, Di
             kwargs:
                 See [`~loaders.LoraLoaderMixin.lora_state_dict`].
         """
-        state_dict, network_alpha = self.lora_state_dict(pretrained_model_name_or_path_or_dict, **kwargs)
-        self.load_lora_into_unet(state_dict, network_alpha=network_alpha, unet=self.unet)
+        state_dict, network_alpha, (unet_state_dict_aux, te_state_dict_aux) = self.lora_state_dict(
+            pretrained_model_name_or_path_or_dict, **kwargs
+        )
+        self.load_lora_into_unet(
+            state_dict, network_alpha=network_alpha, unet=self.unet, state_dict_aux=unet_state_dict_aux
+        )
         self.load_lora_into_text_encoder(
-            state_dict, network_alpha=network_alpha, text_encoder=self.text_encoder, lora_scale=self.lora_scale
+            state_dict,
+            network_alpha=network_alpha,
+            text_encoder=self.text_encoder,
+            lora_scale=self.lora_scale,
+            state_dict_aux=te_state_dict_aux,
```

Comment:
Nice this works for me!

We'd potentially have to revisit this or add a separate `LoraLoaderMixinForSDXL` class to handle SDXL related modifications where the major modification would be the introduction of another text encoder.